examples/components/NER/traininsilo/run.py

```diff
@@ -0,0 +1,483 @@
+import argparse
+import logging
+import sys
+
+from transformers import (
+    DataCollatorForTokenClassification,
+    AutoModelForTokenClassification,
+    AutoTokenizer,
+    get_scheduler,
+)
+from torch.utils.data import DataLoader
+from datasets import load_from_disk
+from torch.optim import AdamW
+from tqdm.auto import tqdm
+import pandas as pd
+import numpy as np
+import evaluate
+import mlflow
+import torch
+
+
+class NERTrainer:
+    def __init__(
+        self,
+        train_data_dir="./",
+        test_data_dir="./",
+        model_path=None,
+        lr=0.01,
+        epochs=1,
+        batch_size=64,
+        experiment_name="default-experiment",
+        iteration_num=1,
+    ):
+        """NER Trainer trains BERT-base model on the MultiNERD dataset.
+
+        Args:
+            train_data_dir(str, optional): Training data directory path
+            test_data_dir(str, optional): Testing data directory path
+            model_path (str, optional): Model path. Defaults to None
+            lr (float, optional): Learning rate. Defaults to 0.01
+            epochs (int, optional): number of epochs. Defaults to 1
+            batch_size (int, optional): DataLoader batch size. Defaults to 64
+            experiment_name (str, optional): Experiment name. Default is default-experiment
+            iteration_num (int, optional): Iteration number. Defaults to 1
+
+        Attributes:
+            model_: Huggingface bert-base pretrained model
+            optimizer_: Stochastic gradient descent
+            train_loader_: Training DataLoader
+            test_loader_: Testing DataLoader
+            labelToId_: label to ID mapping
+            idToLabel_: ID to label mapping
+            metric_: Evaluation metrics
+            device_: Either cpu or gpu
+        """
+
+        # training params
+        self._lr = lr
+        self._epochs = epochs
+        self._batch_size = batch_size
+        self._experiment_name = experiment_name
+        self._iteration_num = iteration_num
+        self._model_path = model_path
+
+        # device
+        self.device_ = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+        # tokenizer
+        model_name = "bert-base-cased"
```

Comment:
This seems to be hardcoded in the pre-processing component too. For me it's okay to merge as it is, but we might want to take a note of that potential slight improvement in the future (define the model type as an input parameter, or something hardcoded in a single location in the submit.py file).

Reply:
Did the required changes to make tokenizer-name and model-name configurable.